homework/src/index.js

```diff
@@ -1,47 +1,113 @@
+        container.innerHTML='';
+        const contributorsURL = repository['contributors_url'];
+        createAppend('p', container, { text: 'Contributions', class:"tableData" });
+        try{
+            const contributors = await fetchJSON(contributorsURL)
+            const contTable = createAppend('table', container);
+            const contTbody = createAppend('tbody', contTable);
+            contributors.forEach(contributor => {
+                const contLink = contributor.html_url;
+                const contImg = contributor.avatar_url;
+                const contName = contributor.login;
+                const contNumber = contributor.contributions;
+                const contTr = createAppend('tr', contTbody);
+                const contributorsLink = createAppend('a', contTr, { href: contLink, target: '_blank', class: 'contributorsLink' });
+                const contTr1 = createAppend('tr', contributorsLink, { id: `${contributor['id']}`, class: 'contributorsTr' });
+                const tdImg = createAppend('td', contTr1, { id: `${contName}Td`});
+                createAppend('img', tdImg, { src: contImg, alt: `picture of ${contName}`, class: 'cont-images' });
+                createAppend('td', contTr1, { text: contName, class: 'cont-info cont-name' });
+                createAppend('td', contTr1, { text: contNumber, class: 'cont-info cont-num' });
+           });
+        }
+        catch(err) {
+           createAppend('div', container, { text: err.message, class: 'alert-error' });
+         }
+    }
+    
+    async function main(url){
+        const result = document.getElementById('result');
+        
+        try{
+            const repositories  = await fetchJSON (url);
+            repositories.sort((a, b) => a.name.localeCompare(b.name));
+            createAppend('div',result,{id:'header'});
+            createAppend('label',header,{text:'HYF Repositories',id:'selectLabel'});
+            createAppend('select',header,{class:'rep-select', id:'selectID'});
+            const select = document.getElementById('selectID');
+            const repoContainer= createAppend('div', result,{ id:'tableContainer'});
+            const contContainer= createAppend('div', result,{ id:'contContainer'});          
+            repositories.forEach((element,index) => { 
+                createAppend('option',select,{text: element.name, value: index}); 
+            });
+            select.addEventListener('change', (event)=> {
+                const change = repositories[event.target.value];
+                renderRepository(repoContainer,change);
+                createContainer(contContainer,change)
+                });
+                renderRepository(repoContainer, repositories[0]);
+                createContainer(contContainer,repositories[0]);
+        }catch(error){
+            createAppend('div', document.getElementById('result'), { text:error.message, class: 'alert-error' });
```

Comment:
You don't need `getElementById` here as the `result` variable is available in the outer scope of the `catch` handler.